app/models/reservation.rb

```diff
@@ -206,6 +206,16 @@ def assign_actuals_off_reserve
     self.actual_end_at   ||= reserve_end_at
   end
 
+  def valid_as_user?(user)
```

Comment:
This is essentially a duplicate of `save_as_user`, but with `valid?`. I might try again to DRY it up, but my initial attempts didn't work.